content/ui-patterns/notification-messaging.mdx

```diff
@@ -0,0 +1,230 @@
+  alt="An upsell banner inside the rulesets setting page on GitHub explaing that this feature can be used if the user upgrades to an Enterprise account."
+  src="https://github.com/primer/design/assets/18661030/924884b8-4b85-4967-898d-e0c2fcfaf785"
+/>
+
+## Placement
+
+Message proximity contributes to the prominence of a message. Whenever possible, place messages near the action they are related to.
+
+### Top of body
+
+If messaging pertains to an entire page or section, place it at the top of the body content. This is the most common placement for messaging components. Banners should fill the container width but maintain appropriate spacing between elements and not appear full width or flush under the global nav.
+
+<img
+  width="960"
+  alt="Highlights a banner at the top of the body inside a GitHub issue."
+  src="https://github.com/primer/design/assets/18661030/73a2ace1-e4fe-4255-850d-062c8d75b7a9"
+/>
+
+### Inline
+
+Inline messages are used to provide feedback on a specific action or input field. They should be placed near the action they are related to.
+
+<img
+  width="960"
+  alt="Highlights an inline message next to an input field inside a GitHub issue."
+  src="https://github.com/primer/design/assets/18661030/ea98bf69-1a7a-4061-8fce-2e2be4303b55"
+/>
+
+### Inside a dialog
+
+Banners should be placed below the Dialog header and appear full width within the Dialog content area. Prioritize showing messaging _inside_ the Dialog after an action rather than closing the Dialog and showing a banner within the page.
+
+<img
+  width="960"
+  alt="Highlights a banner inside a dialog showing an error state has occured."
+  src="https://github.com/primer/design/assets/18661030/f4e3a125-94b6-4b12-b9ec-6b881c25f82c"
+/>
+
+### Placement recommendations
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/user-attachments/assets/47f4fb93-4d79-418c-9868-ec26ecfafdd4" />
+    <Caption>Place banners in close proximity to where the interaction occured.</Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/user-attachments/assets/f4cb187f-3bb4-42de-b901-8cf371371c60" />
+    <Caption>Don't place banners related to internal dialog content at the top of a page.</Caption>
+  </Dont>
+</DoDontContainer>
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/user-attachments/assets/a589e0fe-ca1d-44d3-9f66-bfb2f9d93630" />
+    <Caption>Use InlineMessage when immediate feedback is needed as the result of an action.</Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/user-attachments/assets/de8fcd27-c6ca-4470-8894-bcb9e4373089" />
+    <Caption>Don’t place messaging far from the user action when space is available.</Caption>
+  </Dont>
+</DoDontContainer>
+
+## Choosing a message type
+
+Picking a notification message for an experience can be tricky. Reference this flowchart to help pick the best component for your feature. For more nuanced scenarios, reach out to the Primer team for recommendations.
+
+<img
```

Comment:
The only thing with this one is that it's hard to read and I'm not able to select it to make it bigger in a new window (that might be a limitation of Primer.style?).

Reply:
Spoke on slack; you can open in a new tab with the browser context menu and zoom in 👍